topics/releases.md

```diff
@@ -1,81 +1,103 @@
-===
+### Release Candidate
 
-When a new version of Redis starts to be planned, the unstable branch
-(or sometimes the currently stable branch) is forked into a new
-branch that has the name of the target release.
+New minor and major versions of Redis begin as forks of the `unstable` branch.
+The forked branch's name is the target release
 
-For instance, when Redis 2.6 was released as stable, the `unstable` branch
-was forked into the `2.8` branch.
+For example, when Redis 6.0 was released as stable, the `unstable` branch was forked into the `6.2` branch. The new branch is the release candidate (RC) for that version.
 
-This new branch can be at three different levels of stability:
-development, frozen, and release candidate.
+Bug fixes and new features that can be stabilized during the release's time frame are committed to the release candidate branch.
+That said, the `unstable` branch may include work that is not a part of the release candidate.
 
-* Development: new features and bug fixes are committed into the branch, but not everything going into `unstable` is merged here. Only the features that can become stable in a reasonable time frame are merged.
-* Frozen: no new feature is added, unless it is almost guaranteed to have zero stability impacts on the source code, and at the same time for some reason it is a very important feature that must be shipped ASAP. Big code changes are only allowed when they are needed in order to fix bugs.
-* Release Candidate: only fixes are committed against this release.
+The first release candidate, or RC1, is released once it can be used for development purposes and for testing the new version.
+At this stage, most of the new features and changes the new version brings are ready for review, and the release's purpose is collecting the public's feedback.
 
-Stable tree
-===
+Subsequent release candidates are released every three weeks or so, primarily for fixing bugs.
+These may also add new features and introduce changes, but at a decreasing rate towards the final release candidate.
+
+### Stable Tree
+
+Once development has ended and the frequency of critical bug reports for the release candidate wanes, it is ready for the final release.
+
+At this point, the release is marked as stable.
+
+Versioning
+---
+
+Stable releases liberally follow the usual `major.minor.patch` semantic versioning schema.
+The primary goal is to provide explicit guarantees regarding backward compatibility.
+
+### Patch-Level Versions
 
-At some point, when a given Redis release is in the Release Candidate state
-for enough time, we observe that the frequency at which critical bugs are
-signaled starts to decrease, to the point that for a few weeks we don't have
-any serious bugs reported.
+Patches primarily consist of bug fixes and never introduce any compatibility issues.
 
-When this happens, the release is marked as stable.
+Upgrading from a previous patch-level version is always safe and seamless.
```

Comment:
IIRC upgrading to 6.0.4 (meaningful offset) wasn't always seamless.
not sure we wanna mention that, so your text is ok.

Reply:
Let's hope that was the last time "we" did that.